ex2_services/Services/source/general.c

```diff
@@ -144,28 +141,105 @@ SAT_returnState general_app(csp_packet_t *packet) {
         }
 
         break;
+    }
+
+    case DEPLOY_DEPLOYABLES: {
+        Deployable_t dep;
+        dep = 0;
+        memcpy(&dep, &packet->data[IN_DATA_BYTE], sizeof(uint8_t));
+        uint16_t burnwire_current = 0;
+        status = deploy(dep, &burnwire_current);
+        memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+        memcpy(&packet->data[OUT_DATA_BYTE], &burnwire_current, sizeof(uint16_t));
+        set_packet_length(packet, sizeof(int8_t) + sizeof(uint16_t) + 1); // +1 for subservice
+
+        if (!csp_send(conn, packet, 50)) {
+            csp_buffer_free(packet);
+        }
+
+        break;
+    }
+
+    case GET_SWITCH_STATUS: {
+        uint8_t sw[8] = {0};
+        for (int i = 0; i < 8; i++) {
+            sw[i] = switchstatus(i);
+        }
+        packet->data[STATUS_BYTE] = 0;
+        memcpy(&packet->data[OUT_DATA_BYTE], sw, sizeof(sw));
+        set_packet_length(packet, sizeof(sw) + 2); // +1 for subservice
+
+        if (!csp_send(conn, packet, 50)) {
+            csp_buffer_free(packet);
+        }
+
+        break;
+    }
 
-    case GET_UHF_WATCHDOG_TIMEOUT:
+    case GET_UHF_WATCHDOG_TIMEOUT: {
         status = 0;
         memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
         unsigned int timeout = get_uhf_watchdog_delay();
         memcpy(&packet->data[OUT_DATA_BYTE], &timeout, sizeof(unsigned int));
         set_packet_length(packet, sizeof(int8_t) + sizeof(unsigned int) + 1); // +1 for subservice
         csp_send(conn, packet, CSP_TIMEOUT);
```

Comment:
if cep_send is not successful, the csp buffer should be freed

Reply:
Thanks. Fixed now